esteid/generic.py

```diff
@@ -49,15 +67,20 @@ def destroy_digidoc_session(self):
                 except DigiDocError:
                     pass
 
-            del self.request.session['__ddoc_session']
+            del self.request.session[self.DIGIDOC_SESSION_KEY]
 
         except KeyError:
             pass
 
+    def destroy_digidoc_session_data(self):
+        # clear it just in case and then delete
+        self.request.session[self.DIGIDOC_SESSION_KEY] = {}
```

Comment:
Might be paranoid, but if some part of the code has retrieved the key dict and is holding on to it, then reassigning or deleting will leave the key there.

`self.request.session[self.DIGIDOC_SESSION_KEY].clear()` would clear the data and would propagate to every reference

Reply:
nice catch :+1:, did not think about that angle